engine/kinematics.py

```diff
@@ -114,9 +115,11 @@ def integrate_odom(pose, d, phi):
         new_y = pose[1] + d * math.sin(pose[2])
         new_theta = pose[2]
     else:
-        new_x = pose[0] + (d / phi) * (math.sin(pose[2] + phi) - math.sin(pose[2]))
-        new_y = pose[1] + (d / phi) * (-math.cos(pose[2] + phi) + math.cos(pose[2]))
-        new_theta = (pose[2] + phi) % (2 * math.pi)
+        new_x = pose[0] + (d / phi) * \
+            (math.sin(pose[2] + phi) - math.sin(pose[2]))
+        new_y = pose[1] + (d / phi) * \
+            (-math.cos(pose[2] + phi) + math.cos(pose[2]))
+        new_theta = pose[2] + phi
```

Comment:
if we imagine the robot desired angle set to 0, then if the robot is at theta = 0 and turn left slightly, the angle will be 0 - epsilon, which would have (with the constraint) turned into 2 * pi - epsilon, which is not what we want. This is the same with theta = 2 *pi instead of 0